### PR TITLE
Validate pk in job api url

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
Validate pk in the job api url to prevent SQL Injection

### CODE REVIEW

File: gateway/api/views.py [hunk 1]
```diff
@@ -436,6 +437,13 @@ def get_serializer_class(self):
         return self.serializer_class
 
     def get_queryset(self):
+        pk = self.kwargs.get("pk")
+        if pk and not re.match(
+            "^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
+            pk,
+            re.IGNORECASE,
+        ):
+            return None
```
Comment (psschwei) [inline gateway/api/views.py:446]: we should probably log a warning too if this happens

Comment (akihikokuroda) [reply gateway/api/views.py:450]: Added a log entry

Comment (Tansito) [reply gateway/api/views.py:450]: One question @akihikokuroda , is here where we need to check this? Because we are not using `pk` in the entire method 🤔

Comment (akihikokuroda) [reply gateway/api/views.py:450]: The issue is when the request is like this
```
https://qiskit-serverless-dev.quantum.ibm.com/api/v1/jobs/case+randomblob%281000000%29+when+not+null+then+1+else+1+end+/stop
```
It has a bad string in stead of the job id in the request url.  It causes a database request with the string.  The request that has the pk in the url always executes the `get_queryset` method before it calls the individual method.

Comment (Tansito) [reply gateway/api/views.py:450]: I need to say that I am not being able to replicate it

Comment (akihikokuroda) [reply gateway/api/views.py:450]: I haven't tried it by myself.  The issue says

> The query time is controllable using parameter value [case randomblob(1000000) when not null then 1 else 1 end ], which caused the request to take [334] milliseconds, parameter value [case randomblob(10000000) when not null then 1 else 1 end ], which caused the request to take [2,141] milliseconds, when the original unmodified query with value [id] took [824] milliseconds.

Comment (Tansito) [reply gateway/api/views.py:450]: My doubt here is that it surprises me that django orm doesn't manage this and we have similar end-points that passed the tests so I'm trying to figure out we are doing something that we should not do.

Comment (akihikokuroda) [reply gateway/api/views.py:450]: I've just tried to reproduce this issue but it didn't happen.  So I can not see what happened.  How about putting this fix in and see if the issue happens with it?

Comment (Tansito) [reply gateway/api/views.py:450]: Before merge it let me try to pass another scan Monday with security to confirm it and solve any doubt, Aki.

**Comment by Tansito** [comment]
At the-end I think this PR doesn't apply, I think. Should we close it @akihikokuroda ?

**Comment by akihikokuroda** [comment]
OK. I close

**Pull request closed without merging**